src/MetalPerformanceShaders/MPSKernel.cs

```diff
@@ -20,6 +20,18 @@ public static bool Supports (IMTLDevice device)
 			return MPSSupportsMTLDevice (device == null ? IntPtr.Zero : device.Handle);
 		}
 
+		[DllImport (Constants.MetalPerformanceShadersLibrary)]
+		[Introduced (PlatformName.MacCatalyst, 13, 0)]
+		[TV (12,2), Mac (10,14,4), iOS (12,2)]
+		static extern /* id<MTLDevice> _Nullable */ IntPtr MPSGetPreferredDevice (nuint options);
+		public static IMTLDevice GetPreferredDevice (MPSDeviceOptions options)
```

Comment:
You need to have the attribute on the public API too.

They are needed twice

1. the public API is what the IDE can use to tell people if an API is available in the OS version they target;

2. the internal/native API (like the p/invoke above) is needed by our tools (and tests) so we can ensure they API is _really_ available (because headers are full of _lies_)

Reply:
👍 Yep, I figured as much.

Reply:
👍